bin/mycroft-skill-testrunner

```diff
@@ -20,12 +20,38 @@ DIR="$( dirname "$SOURCE" )"
 # Enter the Mycroft venv
 source "$DIR/../venv-activate.sh" -q
 
+function vktest-clear() {
+    FEATURES_DIR="$DIR/../test/integrationtests/voight_kampff/features"
+    num_feature_files=$(ls $FEATURES_DIR | wc -l) 
+    if [ $num_feature_files -gt "2" ] ; then
+        echo "Removing Feature files"
+        rm ${DIR}/../test/integrationtests/voight_kampff/features/*.feature
+    fi
+    STEPS_DIR="$FEATURES_DIR/steps"
+    num_steps_files=$(ls $STEPS_DIR | wc -l)
+    if [ $num_steps_files -gt "2" ] ; then
+        echo "Removing Custom Step files"
+        TMP_DIR="$STEPS_DIR/tmp"
+        mkdir $TMP_DIR
+        mv "$STEPS_DIR/configuration.py" $TMP_DIR
+        mv "$STEPS_DIR/utterance_responses.py" $TMP_DIR
+        rm ${STEPS_DIR}/*.py
+        mv ${TMP_DIR}/* $STEPS_DIR
+        rmdir $TMP_DIR
```

Comment:
Below is nothing that needs to happen now, just something to consider.

We will likely need to expand the defined steps and will likely add more files here and remembering to update this file list will be tricky.

Can we make a change in the test_setup to prepend a `skill_` to the files it pulls in? or rely upon this being run in a git repo and do a rm $STEPS_DIR/* and then a git checkout of the folder to get back all files?

The first option will likely result in issues of double definitions of steps if there are stepfiles in there now until the files have been manually deleted. Either people will have to delete their skills manually or we can check for the old name and remove it when pulling in the stepfiles from skills.

A third option may be having this script read a "manifest.txt" file in the steps folder listing the files that should remain. since it's in the directory it's easier to remember to update.

Reply:
Yeah, I was thinking about this. At the very least I'll add a comment about what the 2 files should be in the Features dir.

should we be appending a quasi-unique identifier for the skill it came from? eg 
`mycroft-npr-news-skill_filename.py`
Just imagining people calling their custom steps file `steps.py` meaning one would get overwritten.

I'm wary of baking in the expectation of it being a git repo, though it's very likely for anyone doing development.

Is there another way that `test_setup` could track the files it's copying in, then remove each filename from the list as they're confirmed deleted?

The `manifest.txt` seems like a simple middle ground.

Reply:
Yes, adding prefixes has been on my TODO for some time, collisions shouldn't occur